Cyclic TS errors with `hub:db` after nuxt prepare (db types and path aliases)

**Describe the bug**
Hello,

After installing `@nuxthub/core` `0.10.5` there's a confusing cycle of TypeScript errors that keeps alternating between two different problems depending on which command you run.

**Steps to reproduce**
Steps to reproduce the behavior:
1. Clone: https://github.com/Yizack/nuxthub-reproduction
2. Fresh install, run `pnpm i` (which runs prepare) → `vue-tsc -b --noEmit`
   - ❌ `db` auto-import has no types
   - ❌ Non-relative path errors in `.nuxt/tsconfig.app.json` and `.nuxt/tsconfig.shared.json`:
3. Run `pnpm dev` wait, stop it, then run `vue-tsc -b --noEmit` again
   - ✅ `db` is now typed correctly
   - ❌ Non-relative path errors still present
4. Run `pnpm postinstall` (which runs prepare) again → `vue-tsc -b --noEmit`
   - ❌ `db` loses its types again
   - ✅ Non-relative path errors disappear

Edit: I just saw a similar issue #792 I wonder if it fixed the issue with this commit https://github.com/nuxt-hub/core/pull/795 but just opening this issue to check

## Example of Logs

<details>

<summary>Step 2</summary>

```log
npx vue-tsc -b --noEmit
.nuxt/tsconfig.app.json:101:9 - error TS5090: Non-relative paths are not allowed when 'baseUrl' is not set. Did you forget a leading './'?

101         "@nuxthub/db"
 you forget a leading './'?

116         "@nuxthub/db/schema"
            ~~~~~~~~~~~~~~~~~~~~

server/api/users.get.ts:4:21 - error TS7006: Parameter 'user' implicitly has an 'any' type.

4   return users.map((user) => ({
                      ~~~~

.nuxt/tsconfig.shared.json:101:9 - error TS5090: Non-relative paths are not allowed when 'baseUrl' is not set. Did you forget a leading './'?       

101         "@nuxthub/db"
            ~~~~~~~~~~~~~

.nuxt/tsconfig.shared.json:116:9 - error TS5090: Non-relative paths are not allowed when 'baseUrl' is not set. Did you forget a leading './'?       

116         "@nuxthub/db/schema"
            ~~~~~~~~~~~~~~~~~~~~


Found 5 errors.
```

</details>

<details>

<summary>Step 3</summary>

```log
npx vue-tsc -b --noEmit
.nuxt/tsconfig.app.json:101:9 - error TS5090: Non-relative paths are not allowed when 'baseUrl' is not set. Did you forget a leading './'?

101         "@nuxthub/db"
            ~~~~~~~~~~~~~

.nuxt/tsconfig.app.json:116:9 - error TS5090: Non-relative paths are not allowed when 'baseUrl' is not set. Did you forget a leading './'?

116         "@nuxthub/db/schema"
            ~~~~~~~~~~~~~~~~~~~~

.nuxt/tsconfig.shared.json:101:9 - error TS5090: Non-relative paths are not allowed when 'baseUrl' is not set. Did you forget a leading './'?

101         "@nuxthub/db"
            ~~~~~~~~~~~~~

.nuxt/tsconfig.shared.json:116:9 - error TS5090: Non-relative paths are not allowed when 'baseUrl' is not set. Did you forget a leading './'?

116         "@nuxthub/db/schema"
            ~~~~~~~~~~~~~~~~~~~~


Found 4 errors.
```

</details>

<details>

<summary>Step 4</summary>

```log
npx vue-tsc -b --noEmit
server/api/users.get.ts:4:21 - error TS7006: Parameter 'user' implicitly has an 'any' type.

4   return users.map((user) => ({
                      ~~~~

server/api/users.get.ts:4:21 - error TS7006: Parameter 'user' implicitly has an 'any' type.

4   return users.map((user) => ({
                      ~~~~


Found 2 errors.
```

</details>

--- a/src/db/setup.ts
+++ b/src/db/setup.ts
@@ -294,6 +294,8 @@ async function generateDatabaseSchema(nuxt: Nuxt, hub: ResolvedHubConfig) {
           const schemaTypes = await readFile(schemaDtsSource, 'utf-8')
           await writeFile(join(physicalDbDir, 'schema.d.mts'), schemaTypes)
         } catch {
+          // During tests, don't overwrite existing types with stub (preserves dev-generated types)
+          if (nuxt.options.test) return
           // Fallback: create a simple re-export if .d.mts doesn't exist yet
           await writeFile(join(physicalDbDir, 'schema.d.mts'), `export * from './schema.mjs'`)
         }

--- a/src/db/setup.ts
+++ b/src/db/setup.ts
@@ -74,8 +74,8 @@ export async function resolveDatabaseConfig(nuxt: Nuxt, hub: HubConfig): Promise
         }
         break
       }
-      // Cloudflare D1 (production only - dev uses local libsql)
-      if (hub.hosting.includes('cloudflare') && !nuxt.options.dev) {
+      // Cloudflare D1 (production only - dev/prepare uses local libsql)
+      if (hub.hosting.includes('cloudflare') && !nuxt.options.dev && !nuxt.options._prepare) {
         config.driver = 'd1'
         break
       }
@@ -288,40 +288,15 @@ async function generateDatabaseSchema(nuxt: Nuxt, hub: ResolvedHubConfig) {
       try {
         await copyFile(join(nuxt.options.buildDir, 'hub/db/schema.mjs'), join(physicalDbDir, 'schema.mjs'))
 
-        // Try to copy the generated .d.mts file for TypeScript support
-        // The .d.mts is generated in the same directory as schema.mjs
+        // Copy the generated .d.mts file for TypeScript support (overwrites stub from setupDatabaseClient)
         const schemaDtsSource = join(nuxt.options.buildDir, 'hub/db/schema.d.mts')
         try {
           const schemaTypes = await readFile(schemaDtsSource, 'utf-8')
           await writeFile(join(physicalDbDir, 'schema.d.mts'), schemaTypes)
         } catch {
           // Fallback: create a simple re-export if .d.mts doesn't exist yet
-          await writeFile(
-            join(physicalDbDir, 'schema.d.mts'),
-            `export * from './schema.mjs'`
-          )
+          await writeFile(join(physicalDbDir, 'schema.d.mts'), `export * from './schema.mjs'`)
         }
-
-        // Create a minimal package.json for Node.js module resolution
-        const packageJson = {
-          name: '@nuxthub/db',
-          version: '0.0.0',
-          type: 'module',
-          exports: {
-            '.': {
-              types: './db.d.ts',
-              default: './db.mjs'
-            },
-            './schema': {
-              types: './schema.d.mts',
-              default: './schema.mjs'
-            }
-          }
-        }
-        await writeFile(
-          join(physicalDbDir, 'package.json'),
-          JSON.stringify(packageJson, null, 2)
-        )
       } catch (error) {
         log.warn(`Failed to copy schema to node_modules/.hub/: ${error}`)
       }
@@ -542,6 +517,26 @@ export const db: ReturnType<typeof drizzleCore<typeof schema>>
     join(physicalDbDir, 'db.d.ts'),
     physicalDbTypes
   )
+
+  // Create package.json and stub schema files for Node.js module resolution
+  // These are needed during `nuxt prepare` so tsconfig paths resolve correctly
+  const packageJson = {
+    name: '@nuxthub/db',
+    version: '0.0.0',
+    type: 'module',
+    exports: {
+      '.': { types: './db.d.ts', default: './db.mjs' },
+      './schema': { types: './schema.d.mts', default: './schema.mjs' }
+    }
+  }
+  try {
+    await writeFile(join(physicalDbDir, 'package.json'), JSON.stringify(packageJson, null, 2))
+    // Stub schema files (overwritten with actual schema during build)
+    await writeFile(join(physicalDbDir, 'schema.mjs'), 'export {}')
+    await writeFile(join(physicalDbDir, 'schema.d.mts'), 'export {}')
+  } catch (error) {
+    throw new Error(`Failed to create @nuxthub/db package files: ${(error as Error).message}`)
+  }
 
   // Create hub:db alias to @nuxthub/db for backwards compatibility
   nuxt.options.alias!['hub:db'] = '@nuxthub/db'

--- a/src/db/setup.ts
+++ b/src/db/setup.ts
@@ -1,4 +1,4 @@
-import { mkdir, copyFile, writeFile, readFile } from 'node:fs/promises'
+import { mkdir, copyFile, writeFile, readFile, stat } from 'node:fs/promises'
 import chokidar from 'chokidar'
 import { glob } from 'tinyglobby'
 import { join, resolve as resolveFs, relative } from 'pathe'
@@ -274,28 +274,39 @@ async function generateDatabaseSchema(nuxt: Nuxt, hub: ResolvedHubConfig) {
     write: true
   })
 
-  // Build schema types during prepare/dev/build
+  // Build schema types during prepare/dev/build, then copy to node_modules
   nuxt.hooks.hookOnce('app:templatesGenerated', async () => {
+    // Build first
     await buildDatabaseSchema(nuxt.options.buildDir, { relativeDir: nuxt.options.rootDir })
-  })
 
-  // Copy schema to node_modules/@nuxthub/db/ for workflow compatibility
-  nuxt.hooks.hookOnce('app:templatesGenerated', async () => {
+    // Then copy to node_modules/@nuxthub/db/ for workflow compatibility
     const physicalDbDir = join(nuxt.options.rootDir, 'node_modules', '@nuxthub', 'db')
     await mkdir(physicalDbDir, { recursive: true })
 
     try {
       await copyFile(join(nuxt.options.buildDir, 'hub/db/schema.mjs'), join(physicalDbDir, 'schema.mjs'))
 
-      // Copy the generated .d.mts file for TypeScript support (overwrites stub from setupDatabaseClient)
-      const schemaDtsSource = join(nuxt.options.buildDir, 'hub/db/schema.d.mts')
+      // Copy the generated .d.mts file for TypeScript support
+      // Try buildDir first, then fall back to .nuxt (for when buildDir is in .cache during build)
+      const buildDirSource = join(nuxt.options.buildDir, 'hub/db/schema.d.mts')
+      const nuxtDirSource = join(nuxt.options.rootDir, '.nuxt/hub/db/schema.d.mts')
+
+      let schemaTypes: string | undefined
       try {
-        const schemaTypes = await readFile(schemaDtsSource, 'utf-8')
-        await writeFile(join(physicalDbDir, 'schema.d.mts'), schemaTypes)
+        schemaTypes = await readFile(buildDirSource, 'utf-8')
       } catch {
-        // During tests, don't overwrite existing types with stub (preserves dev-generated types)
-        if (nuxt.options.test) return
-        // Fallback: create a simple re-export if .d.mts doesn't exist yet
+        // Fallback to .nuxt directory (types generated during prepare)
+        try {
+          schemaTypes = await readFile(nuxtDirSource, 'utf-8')
+        } catch {
+          // Types not found in either location
+        }
+      }
+
+      if (schemaTypes && schemaTypes.length > 50) {
+        await writeFile(join(physicalDbDir, 'schema.d.mts'), schemaTypes)
+      } else if (!nuxt.options.test) {
+        // Fallback: create a simple re-export if types not available
         await writeFile(join(physicalDbDir, 'schema.d.mts'), `export * from './schema.mjs'`)
       }
     } catch (error) {
@@ -531,9 +542,13 @@ export const db: ReturnType<typeof drizzleCore<typeof schema>>
   }
   try {
     await writeFile(join(physicalDbDir, 'package.json'), JSON.stringify(packageJson, null, 2))
-    // Stub schema files (overwritten with actual schema during build)
-    await writeFile(join(physicalDbDir, 'schema.mjs'), 'export {}')
-    await writeFile(join(physicalDbDir, 'schema.d.mts'), 'export {}')
+    // Stub schema files only if they don't exist (real types written by app:templatesGenerated hook)
+    const schemaPath = join(physicalDbDir, 'schema.mjs')
+    const schemaDtsPath = join(physicalDbDir, 'schema.d.mts')
+    const schemaExists = await stat(schemaPath).then(s => s.size > 20).catch(() => false)
+    const schemaDtsExists = await stat(schemaDtsPath).then(s => s.size > 20).catch(() => false)
+    if (!schemaExists) await writeFile(schemaPath, 'export {}')
+    if (!schemaDtsExists) await writeFile(schemaDtsPath, 'export {}')
   } catch (error) {
     throw new Error(`Failed to create @nuxthub/db package files: ${(error as Error).message}`)
   }

--- a/src/db/setup.ts
+++ b/src/db/setup.ts
@@ -280,30 +280,28 @@ async function generateDatabaseSchema(nuxt: Nuxt, hub: ResolvedHubConfig) {
   })
 
   // Copy schema to node_modules/@nuxthub/db/ for workflow compatibility
-  if (!nuxt.options._prepare) {
-    nuxt.hooks.hookOnce('app:templatesGenerated', async () => {
-      const physicalDbDir = join(nuxt.options.rootDir, 'node_modules', '@nuxthub', 'db')
-      await mkdir(physicalDbDir, { recursive: true })
+  nuxt.hooks.hookOnce('app:templatesGenerated', async () => {
+    const physicalDbDir = join(nuxt.options.rootDir, 'node_modules', '@nuxthub', 'db')
+    await mkdir(physicalDbDir, { recursive: true })
 
+    try {
+      await copyFile(join(nuxt.options.buildDir, 'hub/db/schema.mjs'), join(physicalDbDir, 'schema.mjs'))
+
+      // Copy the generated .d.mts file for TypeScript support (overwrites stub from setupDatabaseClient)
+      const schemaDtsSource = join(nuxt.options.buildDir, 'hub/db/schema.d.mts')
       try {
-        await copyFile(join(nuxt.options.buildDir, 'hub/db/schema.mjs'), join(physicalDbDir, 'schema.mjs'))
-
-        // Copy the generated .d.mts file for TypeScript support (overwrites stub from setupDatabaseClient)
-        const schemaDtsSource = join(nuxt.options.buildDir, 'hub/db/schema.d.mts')
-        try {
-          const schemaTypes = await readFile(schemaDtsSource, 'utf-8')
-          await writeFile(join(physicalDbDir, 'schema.d.mts'), schemaTypes)
-        } catch {
-          // During tests, don't overwrite existing types with stub (preserves dev-generated types)
-          if (nuxt.options.test) return
-          // Fallback: create a simple re-export if .d.mts doesn't exist yet
-          await writeFile(join(physicalDbDir, 'schema.d.mts'), `export * from './schema.mjs'`)
-        }
-      } catch (error) {
-        log.warn(`Failed to copy schema to node_modules/.hub/: ${error}`)
+        const schemaTypes = await readFile(schemaDtsSource, 'utf-8')
+        await writeFile(join(physicalDbDir, 'schema.d.mts'), schemaTypes)
+      } catch {
+        // During tests, don't overwrite existing types with stub (preserves dev-generated types)
+        if (nuxt.options.test) return
+        // Fallback: create a simple re-export if .d.mts doesn't exist yet
+        await writeFile(join(physicalDbDir, 'schema.d.mts'), `export * from './schema.mjs'`)
       }
-    })
-  }
+    } catch (error) {
+      log.warn(`Failed to copy schema to node_modules/.hub/: ${error}`)
+    }
+  })
 
   nuxt.options.alias ||= {}
   // Create hub:db:schema alias to @nuxthub/db/schema for backwards compatibility

